account-management-integration-tests/src/test/java/uk/gov/di/accountmanagement/api/MFAMethodsCreateHandlerIntegrationTest.java

```diff
@@ -91,6 +92,56 @@ void shouldReturn200AndMfaMethodDataWhenAuthAppUserAddsSmsMfa() {
                         .anyMatch(m -> true));
     }
 
+    @Test
+    void shouldReturn200AndMfaMethodDataWhenSmsUserAddsAuthAppMfa() {
+        userStore.addMfaMethodSupportingMultiple(
+                TEST_EMAIL,
+                MFAMethod.smsMfaMethod(
+                        true,
+                        true,
+                        TEST_PHONE_NUMBER,
+                        PriorityIdentifier.DEFAULT,
+                        UUID.randomUUID().toString()));
+        var response =
+                makeRequest(
+                        Optional.of(
+                                constructRequestBody(
+                                        PriorityIdentifier.BACKUP,
+                                        new AuthAppMfaDetail(
+                                                MFAMethodType.AUTH_APP, TEST_CREDENTIAL))),
+                        Collections.emptyMap(),
+                        Collections.emptyMap(),
+                        Map.of("publicSubjectId", TEST_PUBLIC_SUBJECT));
+        assertEquals(200, response.getStatusCode());
+
+        String extractedMfaIdentifier =
+                JsonParser.parseString(response.getBody())
+                        .getAsJsonObject()
+                        .get("mfaIdentifier")
+                        .getAsString();
+
+        List<MFAMethod> mfaMethods = userStore.getMfaMethod(TEST_EMAIL);
+
+        assertTrue(
```

Comment:
I find this quite hard to read as an assertion.

I think I'd rather grab the sms method and then assert on fields, e.g:

```
        var retrievedSmsMethod = mfaMethods.stream().filter(mfaMethod -> mfaMethod.getMfaMethodType().equals(MFAMethodType.SMS.getValue())).findFirst().get();

assertEquals(expectedFoo, retrievedSmsMethod.foo());
etc...

```